tests/views/test_auto_security.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Automatic security tests.
+
+This file isn't meant for tests specific to a particular page; they're designed
+as general-purpose tests for common security requirements. All paths must be
+included in the AutoSecurityTests.PATH_TEST_INFO list (this is tested to ensure
+we can't forget these tests for a new view).
+"""
+
+import collections
+import copy
+import os
+
+import django.urls
+
+import urls
+import utils
+
+import view_tests_base
+
+
+# A tuple containing configuration for automatic security tests for a path.
+PathTestInfo = collections.namedtuple(
+    'PathTestInfo',
+    [
+        # Whether the path is expected to accept GET requests.
+        'accepts_get',
+        # Whether the past is expected to accept POST requests.
+        'accepts_post',
+        # Whether the page is restricted to admins.
+        'restricted_to_admins',
+        # Whether POST requests should require an XSRF token.
+        'requires_xsrf',
```

Comment:
will this every be false if accepts_post is true?

Reply:
Yeah, because we only use XSRF tokens for admin pages. The user-facing pages don't require login, so there's not much point in using XSRF.